### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by tcquinn**
I added various kinds of additional error handling:
* Additional error checking
    - Functions that set configuration can check to see if configuration was successful
    - Function that parses location data checks data length before parsing
* Basic logging
* Retry logic for all functions that communicate with device

For the retry logic, I used a fork of [retrying](https://pypi.org/project/retrying/) called [tenacity](https://github.com/jd/tenacity) because it seems like [retrying](https://pypi.org/project/retrying/) is no longer supported.

### CODE REVIEW

File: decawave_ble/__init__.py [hunk 1]
```diff
@@ -1,6 +1,20 @@
...
+import logging
+
+logger = logging.getLogger(__name__)
+
+retry_exponential_wait_multiplier = 0.1
+retry_num_attempts = 3
+exponential_retry = tenacity.retry(
+        stop = tenacity.stop_after_attempt(retry_num_attempts),
+        wait = tenacity.wait_exponential(multiplier=retry_exponential_wait_multiplier),
+        before = tenacity.before_log(logger, logging.DEBUG),
```
Comment (optimuspaul) [inline decawave_ble/__init__.py:14]: Should make the log level configurable in case debug is too chatty.

**Review by optimuspaul — CHANGES_REQUESTED** [review]
The logger should already have the level set when you config it in your tests, shouldn't need to force the level to DEBUG on the `tenacity` calls.

File: decawave_ble/__init__.py (hunk 1 above)
Comment (tcquinn) [reply decawave_ble/__init__.py:14]: Hmmm...I think I don't understand how the logging works, then. I thought that the parent logger established what to _do_ with the logs of various levels (should we handle them at all, where should we route them, how should we format them, etc.) whereas the log levels within the `tenacity` call are specifying what log level to _assign_ to each type of message from `tenacity`. I assigned `DEBUG` to the `before` and `after` messages precisely to make it easy to ignore them at the parent level.

Comment (tcquinn) [reply decawave_ble/__init__.py:14]: For now, the root loggers in the test functions are, in fact, set to `DEBUG`, but that's just for testing. I assumed that's where we'd change the log level to make it less chatty.

**Pull request merged**